### PR TITLE
Static code analysis using pylint and flake8 - Setting up lint check in CI

### PR DESCRIPTION
**Pull request by atbagga**

### CODE REVIEW

File: .azure-pipelines/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    inputs:
+      versionSpec: '3.x'
+      architecture: 'x64'
+
+  - template: templates/setup-ci-machine.yml
+
+  - template: templates/build-publish-azure-cli-test-sdk.yml
+  
+
+- job: 'Run_Style_Check'
```
Comment (ishitam8) [inline .azure-pipelines/azure-pipelines.yml:44]: Is this for running tests? 
We are anyways running the style check later

Comment (atbagga) [reply .azure-pipelines/azure-pipelines.yml:44]: Sorry didn't get it.. We are running style check in this job only.

Comment (ishitam8) [reply .azure-pipelines/azure-pipelines.yml:44]: Sorry, my bad. I thought it were two separate jobs.

**Review by ishitam8 — APPROVED** [review]

File: deploy-aks/azext_aks_deploy/dev/functionapp/up.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+logger = get_logger(__name__)
+
+
+def functionapp_deploy(repository=None, skip_secrets_generation=False, do_not_wait=False):
+    """Build and Deploy to Azure FunctionApp via GitHub actions
+    :param repository: GitHub repository URL e.g. https://github.com/azure/azure-cli.
+    :type repository: str
+    :param skip_secrets_generation : Flag to skip generating Azure credentials.
+    :type skip_secrets_generation: bool
+    :param no_wait : Do not wait for workflow completion.
```
Comment (ishitam8) [inline deploy-aks/azext_aks_deploy/dev/functionapp/up.py:27]: param name should be : do_not_wait

**Comment by atbagga** [comment]
@ishitam8 I have done sanity check for az aks up. 

Please double check the corner cases whenever you are testing next so as to find any thing this PR might have messed up. (There were lot of merge conflicts for me to resolve.)

**Pull request merged**